app/models/order.rb

```diff
@@ -35,7 +39,15 @@ def set_total
   end
 
   def received
-    update(status: 'received', placed_on: Time.now)
+    update(status: :received, placed_on: Time.now)
+  end
+
+  def cancelled
+    update(status: :cancelled)
+  end
+
+  def cancellable?
```

Comment:
why isn't 30 minutes logic added here?

Reply:
Sir added it here also and it should also be in before_save according to me. Updated